frontend/src/js/common-ui/forms/CountrySelect.tsx

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+import { Controller } from 'react-hook-form';
+
+import { Autocomplete, TextField } from '@mui/material';
+
+import { countries } from '@northern.tech/store/constants';
+
+interface CountrySelectProps {
+  id?: string;
+  onChange: (...event: any[]) => void;
+  [other: string]: any;
+}
+export const CountrySelect = (props: CountrySelectProps) => {
+  const { id, onChange, ...restProps } = props;
+  return (
+    <Autocomplete
+      getOptionLabel={option => option.label}
+      options={countries}
+      sx={{ width: 400 }}
+      autoHighlight
+      renderInput={params => (
+        <TextField
+          {...params}
+          label="Country"
+          id={id || 'country'}
+          slotProps={{
+            htmlInput: {
+              ...params.inputProps,
+              autoComplete: 'new-password' // disable autocomplete and autofill
```

Comment:
This one is from MUI autocomplete example, this way it that doesn't mess with the autofill, as user needs to actually select the country

Reply:
For a `freeSolo` input I could maybe see the option (defaulting to random chars instead of something preselected), for a country selection with a limited selection set I think this can have autocompletion.